lib/split/algorithms/systematic_sampling.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+module Split
+  module Algorithms
+    module SystematicSampling
+      def self.choose_alternative(experiment)
+        block = experiment.cohorting_block
+        raise ArgumentError, "Experiment configuration is missing cohorting_block array" unless block
+        index = experiment.participant_count % block.length
```

Comment:
[Q] Is there any concerns/considerations that need to be taken here with experiment.participant_count and concurrency/race conditions ~ when say 2 accounts are being cohorted at the same time and both retrieve the same participant_count.

Reply:
Yeah, thats definitely a concern, the participant count is incremented in `choose!()` after this algorithm returns a value. I think the solution to this would be to store our own counter field in redis, then at the beginning of the algoritihm we can call [`incrby`](https://redis.io/commands/incrby/) on that values and use the return value from it as the `index`